lib/OpenAPI/Client.pm

```diff
@@ -41,6 +42,20 @@ sub call {
   Carp::croak('[OpenAPI::Client] No such operationId');
 }
 
+sub call_p {
+  my $self = shift;
+  my $promise = Mojo::Promise->new;
+  $self->call(@_, sub {
+    my ($self, $tx) = @_;
+    my $err = $tx->error;
+    return $promise->reject($err->{message}) if $err && !$err->{code};
+    return $promise->reject('WebSocket handshake failed')
+      if $tx->req->is_handshake && !$tx->is_websocket;
```

Comment:
Does OpenAPI support WebSockets..?

Reply:
CPANTesters::API uses it, so I consciously left this in. I believe it's a good idea to keep it.